test/e2e/quota/quota_test.go

```diff
@@ -0,0 +1,339 @@
+			require.NoError(t, err, "error creating APIResourceSchema")
+
+			t.Logf("Creating APIExport")
+			servicesAPIExport := &apisv1alpha1.APIExport{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "services",
+				},
+				Spec: apisv1alpha1.APIExportSpec{
+					LatestResourceSchemas: []string{
+						servicesAPIResourceSchema.Name,
+					},
+				},
+			}
+
+			_, err = kcpClusterClient.Cluster(apiProviderClustername).ApisV1alpha1().APIExports().Create(ctx, servicesAPIExport, metav1.CreateOptions{})
+			require.NoError(t, err, "error creating APIExport")
+
+			t.Logf("Create a binding in the user workspace")
+			binding := &apisv1alpha1.APIBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "services",
+				},
+				Spec: apisv1alpha1.APIBindingSpec{
+					Reference: apisv1alpha1.ExportReference{
+						Workspace: &apisv1alpha1.WorkspaceExportReference{
+							Path:       apiProviderClustername.String(),
+							ExportName: servicesAPIExport.Name,
+						},
+					},
+				},
+			}
+
+			_, err = kcpClusterClient.Cluster(userClusterName).ApisV1alpha1().APIBindings().Create(ctx, binding, metav1.CreateOptions{})
+			require.NoError(t, err)
+
+			t.Logf("Wait for binding to be ready")
+			framework.Eventually(t, func() (bool, string) {
+				binding, err := kcpClusterClient.Cluster(userClusterName).ApisV1alpha1().APIBindings().Get(ctx, binding.Name, metav1.GetOptions{})
+				if err != nil {
+					return false, fmt.Sprintf("failed to list Locations: %v", err)
+				}
+				return conditions.IsTrue(binding, apisv1alpha1.InitialBindingCompleted), fmt.Sprintf("binding not bound: %s", toYaml(binding))
+			}, wait.ForeverTestTimeout, time.Millisecond*100)
+
+			t.Logf("Wait for being able to list Services in the user workspace")
+			framework.Eventually(t, func() (bool, string) {
+				_, err := kubeClusterClient.Cluster(userClusterName).CoreV1().Services("").List(ctx, metav1.ListOptions{})
+				if err != nil {
+					return false, fmt.Sprintf("Failed to list Services: %v", err)
+				}
+				return true, ""
+			}, wait.ForeverTestTimeout, time.Millisecond*100)
+
+			t.Log("Create quota in user workspace")
+			quota := &corev1.ResourceQuota{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "quota",
+				},
+				Spec: corev1.ResourceQuotaSpec{
+					Hard: map[corev1.ResourceName]resource.Quantity{
+						"count/services": resource.MustParse("1"),
+					},
+				},
+			}
+
+			_, err = kubeClusterClient.Cluster(userClusterName).CoreV1().ResourceQuotas("default").Create(ctx, quota, metav1.CreateOptions{})
+			require.NoError(t, err, "error creating quota")
+
+			t.Logf("Waiting for quota to show 0 used Services")
+			framework.Eventually(t, func() (bool, string) {
+				quota, err = kubeClusterClient.Cluster(userClusterName).CoreV1().ResourceQuotas("default").Get(ctx, "quota", metav1.GetOptions{})
+				if err != nil {
+					return false, fmt.Sprintf("Error getting ws quota %s|default/quota: %v", userClusterName, err)
+				}
```

Comment:
shouldn't we fail hard here with `require.NoError`? I cannot see why any error here is acceptable.

Reply:
Will change